AndroidSDKTests/src/test/java/com/leanplum/LeanplumInboxMessageTest.java

```diff
@@ -0,0 +1,109 @@
+package com.leanplum;
+
+import com.leanplum.__setup.AbstractTest;
+import com.leanplum._whitebox.utilities.ResponseHelper;
+import com.leanplum.internal.Constants;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests covering Inbox Messages.
+ * @author Sayaan Saha
+ */
+
+public class LeanplumInboxMessageTest extends AbstractTest {
+  LeanplumInbox leanplumInbox;
+  @Before
+  public void setUp() {
+    leanplumInbox = Leanplum.getInbox();
+    setupSDK(mContext, "/responses/simple_start_response.json");
+  }
+
+  @Test
+  public void testMessageCreate() {
+    Date delivery = new Date(100);
+    Date expiration = new Date(200);
+    HashMap<String, Object> map = new HashMap<>();
+    map.put(Constants.Keys.MESSAGE_DATA, new HashMap<String, Object>());
+    map.put(Constants.Keys.DELIVERY_TIMESTAMP, delivery.getTime());
+    map.put(Constants.Keys.EXPIRATION_TIMESTAMP, expiration.getTime());
+    map.put(Constants.Keys.IS_READ, true);
+
+    LeanplumInboxMessage message = LeanplumInboxMessage.createFromJsonMap("message##Id", map);
+    assertEquals("message##Id", message.getMessageId());
+    assertEquals(delivery, message.getDeliveryTimestamp());
+    assertEquals(expiration, message.getExpirationTimestamp());
+    assertTrue(message.isRead());
+    assertNull(message.getData());
+
+    assertNull(message.getImageFilePath());
+    assertNull(message.getImageUrl());
+  }
+
+  @Test
+  public void testReadAndUpdateMessageCount() {
+    Date delivery = new Date(100);
+    Date expiration = new Date(200);
+    HashMap<String, Object> map = new HashMap<>();
+    map.put(Constants.Keys.MESSAGE_DATA, new HashMap<String, Object>());
+    map.put(Constants.Keys.DELIVERY_TIMESTAMP, delivery.getTime());
+    map.put(Constants.Keys.EXPIRATION_TIMESTAMP, expiration.getTime());
+    map.put(Constants.Keys.IS_READ, false);
+    LeanplumInboxMessage message = LeanplumInboxMessage
+        .createFromJsonMap("messageId##00", map);
+
+    assertEquals(false, message.isRead());
+
+    message.read();
+
+    assertEquals(true, message.isRead());
```

Comment:
is this test implemented here https://github.com/Leanplum/Leanplum-Android-SDK/pull/164/files#diff-d4fac6525f945e4a092843223e65630dR131 ?

Reply:
I'm pulling it out of testInbox since that test is way to convoluted.

Reply:
why do we need to test twice the same?

Reply:
Agreed, I really want to delete the one from testInbox().